pcsx2/DEV9/ATA/ATA_Transfer.cpp

```diff
@@ -235,8 +235,6 @@ void ATA::IO_SparseCacheLoad()
 		hddSparseBlockValid = true;
 #if defined(PCSX2_DEBUG) || defined(PCSX2_DEVBUILD)
 
-		// Store file pointer.
-		const s64 orgPos = FileSystem::FTell64(hddImage);
 		pxAssert(orgPos != -1);
```

Comment:
the fetching orgPos was there to fire the asserts, so we knew if the hddImage was not found. So either get rid of the assert or put it back, either or.

Reply:
This line also caught my attention, and had to double check this.

Yes I added this for an assert, but this definition of orgPos ends up shadowing another definition of orgPos a few lines up (which should already have the same value)

Reply:
Cool then in that case the assert can be moved up (unless there's one already there, don't have the code open) and the others removed :)

Reply:
The assert should only run when the platform specific file apis (`DeviceIoControl` or `lseek`) say that part of the file is sparse, That wouldn't be the case if moved up.

That being said, I'm sure this assert could still be deduplicated

Reply:
Why not just `pxAssert(FileSystem::FTell64(hddImage) != -1);`?

FTell64 won't have any side-effects on the stream, so the lack of call in release should be fine.

Reply:
All good now or you want the assert placed in a different place?

Reply:
Ugh, so you are keeping the variable around. So then you should assert on the variable, not call it again (I assumed the variable was being removed).

The whole thing's a bit iffy tbh. Later in the function, it has error handling for the tell fail case (but tell should generally never fail), but also asserts on it...

Reply:
Yes we are keeping it, I found it odd as well that you suggested the above change, anyway should be fine now.